### PR TITLE
fix(core):  unset objects when no values in patch when empty value

### PR DESCRIPTION
**Pull request by RitaDias**
### Description

A bug was reported (like #4285) where objects weren't being unset when the values were removed. Now the object input will remove itself completely from a patch if it identifies that it should

### What to review

- [On simple object (Obj Name), removing the value unsets the object input](https://test-studio-git-bug-sc-32671.sanity.build/test/content/input-debug;empty;85909e44-4e17-4dd1-adf3-117681e59a7e)
- [On object with multiple values (Object with columns), removing one value unsets only the removed value](https://test-studio-git-bug-sc-32671.sanity.build/test/content/input-standard;objectsTest;91476f47-f035-47d7-8577-eba51f33964f%2Cpath%3DobjectWithColumns.string1)
- [On object with multiple values (Object with columns), removing all values unsets the whole field](https://test-studio-git-bug-sc-32671.sanity.build/test/content/input-standard;objectsTest;91476f47-f035-47d7-8577-eba51f33964f%2Cpath%3DobjectWithColumns.string1)
- [On custom object field (MyObject), removing one value unsets only the removed value](https://test-studio-git-bug-sc-32671.sanity.build/test/content/input-standard;objectsTest;91476f47-f035-47d7-8577-eba51f33964f%2Cpath%3DmyObject.first)
- [On custom object field (MyObject), removing all values unsets the whole field](https://test-studio-git-bug-sc-32671.sanity.build/test/content/input-standard;objectsTest;91476f47-f035-47d7-8577-eba51f33964f%2Cpath%3DmyObject.first)
- [On deeply nested object (recursive), removing separate values should have no impact on the rest](https://test-studio-git-bug-sc-32671.sanity.build/test/content/input-standard;objectsTest;91476f47-f035-47d7-8577-eba51f33964f%2Cpath%3Drecursive.objectWithColumns.string1)
- [On deeply nested object (recursive), removing all values should have no impact on the rest](https://test-studio-git-bug-sc-32671.sanity.build/test/content/input-standard;objectsTest;91476f47-f035-47d7-8577-eba51f33964f%2Cpath%3Drecursive.objectWithColumns.string1)

### Notes for release

Fixes an issue where an object field’s value is not set to undefined when the last item is removed

### CODE REVIEW

**Comment by binoy14** [comment]
Updated description to use preview URL instead of localhost

**Review by binoy14 — APPROVED** [review]
Looks good to me, thanks for tackling this!

File: packages/sanity/src/core/form/members/object/fields/ObjectField.tsx
Comment (rexxars) [inline packages/sanity/src/core/form/members/object/fields/ObjectField.tsx:70]:
```suggestion
      // if the patch is an unset patch that targets a field in the object (as opposed to unsetting a field somewhere deeper)
```

File: packages/sanity/src/core/form/members/object/fields/ObjectField.tsx
Comment (rexxars) [inline packages/sanity/src/core/form/members/object/fields/ObjectField.tsx:85]: I’m not sure I see how this comment relates here, since we’re not checking for the value of the type - but don’t let that block the PR. Would be nice to clarify the purpose of this path after merging though.

**Review by rexxars — APPROVED** [review]

File: packages/sanity/src/core/form/members/object/fields/ObjectField.tsx
Comment (RitaDias) [reply packages/sanity/src/core/form/members/object/fields/ObjectField.tsx:85]: This comment was originally one that we had in the old code of v2. 
Let me clarify it though :)

Comment (RitaDias) [reply packages/sanity/src/core/form/members/object/fields/ObjectField.tsx]: I've updated it, hopefully it's a bit clearer why it exists 👍

**Pull request merged**
